test/controllers/admin/comfy/external_api_clients_controller_test.rb

```diff
@@ -17,6 +17,43 @@ class Comfy::Admin::ExternalApiClientsControllerTest < ActionDispatch::Integrati
     @external_api_client.update(status: ExternalApiClient::STATUSES[:stopped], enabled: true)
     get start_api_namespace_external_api_client_path(api_namespace_id: @api_namespace.id, id: @external_api_client.id)
     assert_response :redirect
+    assert_not_empty ExternalApiClientJob.jobs
+    Sidekiq::Worker.drain_all
+  end
+
+  test "#start: should not run sidekiq-job if external-api-client is disabled" do
+    @external_api_client.update(status: ExternalApiClient::STATUSES[:stopped], enabled: false)
+    
+    sign_in(@user)
+    get start_api_namespace_external_api_client_path(api_namespace_id: @api_namespace.id, id: @external_api_client.id)
+    
+    assert_response :redirect
+    assert_empty ExternalApiClientJob.jobs
+  
+    Sidekiq::Worker.drain_all
+  end
+
+  test "#start: should not run sidekiq-job if external-api-client is in error status" do
+    @external_api_client.update(status: ExternalApiClient::STATUSES[:error], enabled: true)
+    
+    sign_in(@user)
+    get start_api_namespace_external_api_client_path(api_namespace_id: @api_namespace.id, id: @external_api_client.id)
+    
+    assert_response :redirect
+    assert_empty ExternalApiClientJob.jobs
+  
+    Sidekiq::Worker.drain_all
+  end
+
+  test "#start: should not run sidekiq-job if external-api-client is in running status" do
+    @external_api_client.update(status: ExternalApiClient::STATUSES[:running], enabled: true)
+    
```

Comment:
wish we had this test case before the issue that led to [this RCA](https://github.com/restarone/violet_rails/issues/478) @Pralish